pkg/postage/service.go

```diff
@@ -193,7 +194,57 @@ func (ps *service) HandleStampExpiry(id []byte) {
 }
 
 // SetExpired removes all expired batches from the stamp issuers.
-func (ps *service) SetExpired() error {
+func (ps *service) SetExpired(ctx context.Context) error {
+
+	ps.logger.Debug("removing expired stamp data from stamperstore. this may take a while if the node has not been restarted in a while.")
+
+	deleteItemC := make(chan *StampItem)
+	go func() {
+		for item := range deleteItemC {
+			_ = ps.store.Delete(item)
+		}
+	}()
+
+	go func() {
+		count := 0
+		defer func() {
+			close(deleteItemC)
+			ps.logger.Debug("removed expired stamps", "count", count)
+		}()
+
+		err := ps.store.Iterate(
+			storage.Query{
+				Factory: func() storage.Item {
+					return new(StampItem)
+				},
+			}, func(result storage.Result) (bool, error) {
+				item := result.Entry.(*StampItem)
+				exists, err := ps.postageStore.Exists(item.BatchID)
+				if err != nil {
+					return false, fmt.Errorf("set expired: checking if batch exists for stamp item %s: %w", hex.EncodeToString(item.BatchID), err)
+				}
+				if !exists {
+					count++
+
+					select {
+					case deleteItemC <- item:
+					case <-ctx.Done():
+						return false, ctx.Err()
+					}
+
+					if count%100_000 == 0 {
```

Comment:
Where does 100k come from?

Reply:
a random number used ask a checkpoint to show the user that the process is on going